### PR TITLE
xpm: Fix various parser panics

### PR DESCRIPTION
**Pull request by gnoack**
This series of patches adds subtests to test input files against golden PNGs, a fuzzer, and then various changes to fix the detected panics in the parser.

### CODE REVIEW

File: go.mod
Comment (Jacalz) [inline go.mod]: The upcoming version of Fyne is targeting 1.19 as baseline. You might as well bump it once more.

File: xpm/xpm.go
Comment (Jacalz) [inline xpm/xpm.go]: You are not using any formatting. Consider using `errors.New()` instead. You might also want to bring it out to a global error variable given that the same error is used in multiple places.

File: xpm/xpm.go
Comment (Jacalz) [inline xpm/xpm.go]: Consider using `errors.New()`.

File: xpm/xpm.go
Comment (Jacalz) [inline xpm/xpm.go]: Consider using `errors.New()`.

File: xpm/xpm.go
Comment (Jacalz) [inline xpm/xpm.go]: Consider using `errors.New()`.

File: xpm/xpm.go
Comment (Jacalz) [inline xpm/xpm.go]: Consider using `errors.New()`.

**Review by Jacalz — CHANGES_REQUESTED** [review]
Thanks. A very nice change. Added some notes  inline. Mostly about error handling.

File: xpm/fuzz_test.go
Comment (Bluebugs) [inline xpm/fuzz_test.go:14]: I think it would be ok to run fuzz automatically everyday in a github action. I usually pick 4am as that's when the activity on fyne repository are at their lowest. If you do not mind adding a new action, I think that will be useful to run this fuzzing regularly.

File: go.mod
Comment (gnoack) [reply go.mod]: Makes sense, will add a commit for that which you can squash with 5bdced54

File: xpm/xpm.go
Comment (gnoack) [reply xpm/xpm.go]: I introduced an `ErrInvalidFormat` and return `fmt.Errorf("%w: ...")` in the error cases, so that callers can tell it apart.

File: xpm/xpm.go
Comment (gnoack) [reply xpm/xpm.go]: Using `fmt.Errorf("%w: ...", ErrInvalidFormat)` (see comment above)

File: xpm/xpm.go
Comment (gnoack) [reply xpm/xpm.go]: Using `fmt.Errorf("%w: ...", ErrInvalidFormat)` (see comment above)

File: xpm/xpm.go
Comment (gnoack) [reply xpm/xpm.go]: Using `fmt.Errorf("%w: ...", ErrInvalidFormat)` (see comment above)

File: xpm/fuzz_test.go
Comment (gnoack) [reply xpm/fuzz_test.go:14]: Fuzzing runs indefinitely if you let it, and I'm not sure whether Github actions are the best fit there.  (TBH, I'm not terribly familiar with Github actions either.)  I suspect that these considerations would inflate the scope of this change too much.  I'd rather leave it up to you to configure this, if that's OK with you. :)

**Comment by gnoack** [comment]
> Thanks. A very nice change. Added some notes inline. Mostly about error handling.

Thanks for the review. I added some commits on top to address these; you might want to squash the go.mod change onto 5bdced54 before merging.  Please take another look.

**Review by andydotxyz — APPROVED** [review]
Nice addition of safety thanks :)

**Review by Jacalz — APPROVED** [review]

**Comment by Jacalz** [comment]
Thanks for working on this :)

**Pull request merged**
